exonum/src/node/mod.rs

```diff
@@ -760,10 +762,28 @@ impl fmt::Debug for ApiSender {
     }
 }
 
+fn deserialize_socket_address<'de, D>(value: D) -> Result<SocketAddr, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let address_str: String = Deserialize::deserialize(value)?;
+    address_str
+        .to_socket_addrs()
```

Comment:
Question: Hmm.. Shouldn't we warn users that if they use hostnames, they should properly configure DNS and use only trusted (and not the same) servers? In a theoretical scenario, if some common DNS server will be hacked, and (1/3)+ nodes use this server, it may lead to a silent byzantine attack.

Reply:
What do you propose ? Just panic ? :-)

Reply:
I believe it is reasonable to warn the users about the possibility of DNS spoofing somewhere in the documentation on node configuration. But I do not think that we should do anything on nodes about that.

Proper configuration of DNS resolver (a system service) is responsibility of the administrator of node's machine. Using trusted DNS connections is in the same league as expecting that your ISP connects the node to the real Internet and does not reroute IP packets to compromised nodes.

Reply:
Yep, I was talking about some kind of disclaimer.

Reply:
And what if user made a mistake in hostname ? How to inform him ?

Reply:
DNS spoofing, and other kind of traffic-redirection wouldn't lead to "byzantine attack", but can lead to DoS for exonum network.

My opinion, that user story should be next:
1) User sets peers address (it does not matter hostname or ip)
2) Current node trying to connect (trying to resolve hostname, checks pk validity) with some peer.
3) If current node fails to connect to some of peer (hostname/ip wrong or PK is differ), warn should apear in log.
4) If current node fails to connect to (majority_count - 1) peers, error should apear in log.